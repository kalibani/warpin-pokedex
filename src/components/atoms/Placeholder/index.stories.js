// Stories Placeholder Component
// --------------------------------------------------------
import React, { Fragment } from 'react';
import { storiesOf } from '@storybook/react';
import Placeholder from './index';
import Notes from './readme.md';

storiesOf('Atoms', module).add('Placeholder', () => (
  <Placeholder />
), {
  info: { inline: true, header: false, text: Notes }
});
