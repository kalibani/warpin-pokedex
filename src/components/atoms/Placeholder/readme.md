## Placeholder Component
Placeholder Component is ....