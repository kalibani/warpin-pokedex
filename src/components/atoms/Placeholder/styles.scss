/* ==========================================================================
Styles for Card component
========================================================================== */

// import variables
@import "../../../assets/styles/variables";


// Start styles Card component
.a-placeholder {
  margin: 30px 19px;
  padding: 15px;
  border-radius: 10px;
  min-height: calc(180px * 1.5);
  width: 16%;
  overflow: hidden;
  background: linear-gradient(to right, rgba(3, 19, 65, 0.52) 0%, rgba(38, 3, 65, 0.08) 100%);
  animation: pulse 5s infinite;

  @media (min-width: 320px) and (max-width: 480px) {
    flex: 25%;
  }
  

  @keyframes pulse {
    0% {
      background-color: rgba(3, 19, 65, 0.20);
    }
    100% {
      background-color: rgba(158, 152, 163, 0.08);
    }
  }
}
