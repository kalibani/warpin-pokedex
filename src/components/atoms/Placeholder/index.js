// Skeleton Component
// --------------------------------------------------------

import React from 'react';
import { RectShape } from 'react-placeholder/lib/placeholders';
import 'react-placeholder/lib/reactPlaceholder.css';

// styles
import './styles.scss';

const Skeleton = () => (
  <>
    <div className="container">
      <div className="row">
        <div className="a-placeholder">
          <RectShape />
        </div>
        <div className="a-placeholder">
          <RectShape />
        </div>
        <div className="a-placeholder">
          <RectShape />
        </div>
        <div className="a-placeholder">
          <RectShape />
        </div>
        <div className="a-placeholder">
          <RectShape />
        </div>
      </div>
    </div>
  </>
);
export default Skeleton;
