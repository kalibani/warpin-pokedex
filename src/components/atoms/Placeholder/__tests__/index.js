// Test Placeholder Component
// --------------------------------------------------------

import React from 'react';
import Placeholder from '../../index';
import renderer from 'react-test-renderer';

test('Placeholder is match the snapshot', () => {
const component = renderer.create(
Placeholder,
);
let tree = component.toJSON();
expect(tree).toMatchSnapshot();
});
