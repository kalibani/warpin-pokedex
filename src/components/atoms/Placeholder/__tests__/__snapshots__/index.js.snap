// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Placeholder is match the snapshot 1`] = `null`;
